ElectronClient/app.js

```diff
@@ -1396,6 +1394,27 @@ class Application extends BaseApplication {
 			bridge().openDevTools();
 		}
 
+		KeymapService.setLogger(reg.logger());
+
+		const keymapPath = `${dir}/keymap-desktop.json`;
+
+		if (await fs.pathExists(keymapPath)) {
+			try {
+				let keymapFile = await fs.readFile(keymapPath, 'utf-8');
+				keymapFile = keymapFile.replace(/^\s*\/\/.*/, ''); // Strip-off comments
+
+				const customKeymap = JSON.parse(keymapFile);
+				await KeymapService.instance().setKeymap(customKeymap);
+			} catch (error) {
+				const msg = error.message ? error.message : '';
+				console.error(`Failed to parse keymap ${keymapPath}\n${msg}`);
+			}
+		}
```

Comment:
All this should be part of KeymapService.

Let's not support comments in the JSON file for now - it will have to be either valid JSON or it's an error. Also if you strip off comments, I guess you should put them back when the file is modified from the editor and it's getting needlessly complicated. I suppose later on we could use a more advanced JSON parser to support comments, trailing commas, etc. but for now basic JSON.parse is fine.